ssc/cmod_pvsamv1.cpp

```diff
@@ -3158,10 +3213,14 @@ void cm_pvsamv1::exec( ) throw (compute_module::general_error)
 	if (annual_dc_net > 0) percent = 100 * annual_inv_pntloss / annual_dc_net;
 	assign("annual_ac_inv_pnt_loss_percent", var_data((ssc_number_t)percent));
 
+	percent = 0.;
+	if (annual_dc_net > 0) percent = 100 * annual_inv_tdcloss / annual_dc_net;
+	assign("annual_ac_inv_tdc_loss_percent", var_data((ssc_number_t)percent));
+
 	sys_output = annual_dc_net;
-	sys_output -= (annual_inv_cliploss + annual_inv_pntloss + annual_inv_psoloss);
+	sys_output -= (annual_inv_cliploss + annual_inv_pntloss + annual_inv_psoloss + annual_inv_tdcloss);
 	percent = 0.;
-	if (sys_output > 0) percent = 100 * (sys_output - annual_ac_gross) / sys_output;
+	if (annual_dc_net > 0) percent = 100 * (annual_dc_net - sys_output) / annual_dc_net;
```

Comment:
redefine "annual_ac_inv_eff_loss_percent" 

in a test, I derated the inverter to ~50% for 50% of the year but this value only showed 8% efficiency loss, whereas with new definition it shows 20% efficiency loss

Reply:
I think we could modify the battery model (DC-connected) to also derate the inverter for temperature, but we can wait to do that while we focus on this initial integration

Reply:
This change should be reverted- this definition is basically the efficiency of converting from DC to AC, not including clipping losses, nighttime losses, or power draw losses, which is why the annual_ac_gross term needs to be in there. Since we'll want to show inverter thermal losses as their own loss stream separate from the inverter efficiency, line 3221 is correct.

Definitely not the most transparent calculation in the world, and a big area we'd like to fix with some code rewriting. :)

Reply:
reverted!